libqpdf/QPDF.cc

```diff
@@ -1290,65 +1292,48 @@ QPDF::showXRefTable()
     }
 }
 
+// Ensure all objects in the pdf file, including those in indirect references,
+// appear in the object cache.
 void
 QPDF::fixDanglingReferences(bool force)
 {
-    if (this->m->fixed_dangling_refs && (!force)) {
```

Comment:
Why did you remove `!force`? Even though you clear `fixed_dangling_refs`, there may be a reason to call this with `force`. For example, new ones may have come across when copying foreign objects or with other manual manipulation. `QPDFWriter` calls this with force, so we should still respect `force`.

Reply:
My mistake. I took it out for testing and forgot to put it back in.

That said, there probably should be a test case that relies on force == true. Not sure on how to construct it

Reply:
Now fixed.

Reply:
> That said, there probably should be a test case that relies on force == true. Not sure on how to construct it

There probably should be. I'm not sure it's actually possible to create such a case anymore. I can think of ways that would have worked to reproduce it before the object/value split such as inserting a new object with the version of parse that takes a QPDF context and including a dangling reference there, but that would no longer cause a dangling reference because the dangling reference would get inserted into the object cache.

This may just be defensive coding now.